### PR TITLE
Modify tests to work with svn 1.14

### PR DESCRIPTION
**Pull request by dpmatthews**
Also

- Replace Ubuntu 18.04 with 22.04 in github actions
- Skip tests which are broken with svn 1.13.0 (Ubuntu 20.04)
- Use Python 3 (not 2) in tests

### CODE REVIEW

**Comment by dpmatthews** [comment]
Ubuntu 22.04 uses svn 1.14.1 so shouldn't be affected by the svn bug which causes failures with 20.04 (see #267)

**Comment by dpmatthews** [comment]
svn 1.11+ contains improvements to the handling of tree conflicts.
https://subversion.apache.org/docs/release-notes/1.14.html#conflict-resolver
Some of the tests which previously resulted in a tree conflict now just result in a text conflict. The tests have been adapted to reflect this.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -5,12 +5,12 @@ jobs:
     strategy:
       fail-fast: False
       matrix:
-        ubuntu: ['ubuntu-18.04', 'ubuntu-20.04']
+        ubuntu: ['ubuntu-20.04', 'ubuntu-22.04']
```
Comment (wxtim) [inline .github/workflows/tests.yaml:8]: Why not ubuntu-latest?

Sadly, there isn't the option of doing `['ubuntu-earliest', 'unbuntu-second-latest', 'ubuntu-latest']

Comment (dpmatthews) [reply .github/workflows/tests.yaml:8]: No particular reason - I just prefer to be explicit

File: t/lib/bash/test_header [hunk 2]
```diff
@@ -234,13 +251,13 @@ function commit_sort() {
...
     # Sort the 'Adding/Deleting', etc part of the message
-    python2 -c 'import re, sys
+    python3 -c 'import re, sys
 text = sys.stdin.read()
 sending_lines = re.findall("^\w+ing  +.*$", text, re.M)
 prefix = text[:text.index(sending_lines[0])]
 suffix = text[(text.index(sending_lines[-1]) + len(sending_lines[-1])):]
 sending_lines.sort()
-print prefix + "\n".join(sending_lines) + suffix.rstrip()
+print (prefix + "\n".join(sending_lines) + suffix.rstrip())
```
Comment (wxtim) [inline t/lib/bash/test_header:260]: The space between `print` and `(` but I was slightly surprised it was valid. It's unusual.

File: t/lib/bash/test_header [hunk 3]
```diff
@@ -234,13 +251,13 @@ function commit_sort() {
     # Sort the svn status part of the message
     status_sort $INPUT_FILE $TMP_OUTPUT_FILE
     # Sort the 'Adding/Deleting', etc part of the message
-    python2 -c 'import re, sys
+    python3 -c 'import re, sys
 text = sys.stdin.read()
 sending_lines = re.findall("^\w+ing  +.*$", text, re.M)
```
Comment (wxtim) [inline t/lib/bash/test_header:256]: You'll get syntax deprecation warnings with this. 

```
sed -i 's@(re\..*\()"@$1r"@g' t/lib/bash/test_header
```

**Review by wxtim — COMMENTED** [review]
## Python 2→3

In descending order of importance:

- `re.????("...` → `re.????(r"...`: Important if you don't want warning messages.
- `print ("` → `print("`: not an essential change, but it looks odd to my eye.
- Investigated that `list.sort()` behaves the same way for the Python 3 (just in case).

## SVN tests

I have had a skim read through the `*.t` files. I'm not super familiar with FCM/SVN output: I can do a more detailed review if you want, but that would involve considerable learning, and I think will hit the law of diminishing returns.

**Review by wxtim — APPROVED** [review]

**Pull request merged**